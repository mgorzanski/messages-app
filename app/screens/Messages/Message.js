import React from 'react';
import { StyleSheet } from 'react-native';
import * as globalStyles from './../../styles/globalStyles';
import { ListItem, Text, Body, Left, Right } from 'native-base';
import AsyncImage from './../../components/AsyncImage';

export default class Message extends React.PureComponent {
    constructor(props) {
        super(props);
        this.state = {
            render: false
        }
    }

    componentDidMount() {
        setTimeout(() => {this.setState({render: true})}, 100);
    }

    render() {
        const render = this.state.render;

        if (render) {
        return (
            <ListItem avatar onPress={() => this.props.navigation.navigate('MessageThread', {name: this.props.name})}>
                <Left>
                    <AsyncImage source={require('./../../img/icons/user2.png')} style={styles.thumbnail} placeholderColor={globalStyles.$appBackgroundColor} />
                </Left>
                <Body>
                    <Text style={styles.title}>{this.props.name}</Text>
                    {/*
                        If this.props.message was passed, render the text. Otherwise don't render anything here (null).
                    */}
                    { this.props.message ? (
                        <Text style={styles.message} note>{this.props.message}</Text>
                    ) : null}
                </Body>
                <Right>
                    {/*
                        If this.props.date was passed, render the text. Otherwise don't render anything here (null).
                    */}
                    { this.props.date ? (
                        <Text style={styles.timestamp} note>{this.props.date}</Text>
                    ) : null}
                </Right>
            </ListItem>
        );
        }
        return (null);
    }
}

const styles = StyleSheet.create({
    title: {
        color: globalStyles.$white
    },
    message: {
        color: globalStyles.$darkGray,
        fontSize: 13
    },
    timestamp: {
        color: globalStyles.$darkGray,
        fontSize: 13
    },
    thumbnail: {
        width: 50,
        height: 50
    }
});