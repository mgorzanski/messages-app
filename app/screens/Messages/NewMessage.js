import React from 'react';
import * as globalStyles from './../../styles/globalStyles';
import { StyleSheet } from 'react-native';
import { Container, Item, Input, List, ListItem, Left, Right, Text, Header, Content } from 'native-base';
import Icon from './../../utils/Icon';

export default class NewMessage extends React.Component {
    static navigationOptions = {
        title: "New message",
        headerTintColor: globalStyles.$white
    };

    constructor(props) {
        super(props);
        this.state = {
            render: false
        }
    }

    componentDidMount() {
        setTimeout(() => {this.setState({render: true})}, 50);
    }

    render() {
        const render = this.state.render;
        if (render) {
            return (
                <Container>
                    <Header searchBar rounded style={styles.header} androidStatusBarColor={globalStyles.$headerBackgroundColor}>
                        <Item>
                            <Icon family="MaterialIcons" name="search" />
                            <Input placeholder="Search contacts" />
                        </Item>
                    </Header>
                    <Content>
                        <List>
                            <ListItem>
                                <Left>
                                    <Text>Simon Mignolet</Text>
                                </Left>
                                <Right>
                                    <Icon family="MaterialIcons" name="arrow-forward" />
                                </Right>
                            </ListItem>
                            <ListItem>
                                <Left>
                                    <Text>Nathaniel Clyne</Text>
                                </Left>
                                <Right>
                                    <Icon family="MaterialIcons" name="arrow-forward" />
                                </Right>
                            </ListItem>
                            <ListItem>
                                <Left>
                                    <Text>Dejan Lovren</Text>
                                </Left>
                                <Right>
                                    <Icon family="MaterialIcons" name="arrow-forward" />
                                </Right>
                            </ListItem>
                        </List>
                    </Content>
                </Container>
            );
        }
        return (null);
    }
}

const styles = StyleSheet.create({
    header: {
        backgroundColor: globalStyles.$headerBackgroundColor,
        height: 64
    }
});