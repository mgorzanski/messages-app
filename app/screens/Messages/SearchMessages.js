import React from 'react';
import * as globalStyles from './../../styles/globalStyles';
import PropTypes from 'prop-types';
import { StyleSheet } from 'react-native';
import { Container, Item, Input, Button, Text, Header, Content } from 'native-base';
import Icon from './../../utils/Icon';
import AfterInteractions from './../../components/AfterInteractions';

export default class SearchMessages extends React.Component {
    static navigationOptions = {
        header:  null
    };

    render() {
        return (
            <AfterInteractions>
                <Container>
                    <Header searchBar rounded style={styles.header} androidStatusBarColor={globalStyles.$headerBackgroundColor}>
                        <Item>
                            <Icon family="MaterialIcons" name="search" />
                            <Input placeholder="Search messages" />
                        </Item>
                    </Header>
                    <Content style={styles.content}>
                        <Button full dark onPress={() => this.props.navigation.goBack()}>
                            <Text>Back</Text>
                        </Button>
                    </Content>
                </Container>
            </AfterInteractions>
        );
    }
}

const styles = StyleSheet.create({
    content: {
        margin:10
    },
    header: {
        backgroundColor: globalStyles.$headerBackgroundColor,
        height: 64
    }
});

SearchMessages.propTypes = {
    navigation: PropTypes.object.isRequired
};