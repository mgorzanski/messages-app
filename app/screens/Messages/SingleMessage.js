import React from 'react';
import { Text } from 'native-base';
import { View, StyleSheet } from 'react-native';
import PropTypes from 'prop-types';
import * as globalStyles from './../../styles/globalStyles';
import AsyncImage from './../../components/AsyncImage';

const styles = StyleSheet.create({
    otherMessageRow: {
        display: 'flex',
        flexDirection: 'row'
    },
    myMessageRow: {
        display: 'flex',
        flexDirection: 'row-reverse',
        justifyContent: 'flex-start'
    },
    otherMessageViewText: {
        backgroundColor: globalStyles.$otherMessageTextBackground,
        padding: 10,
        marginBottom: 10,
        marginLeft: 10,
        marginRight: 10,
        borderRadius: 5
    },
    myMessageViewText: {
        backgroundColor: globalStyles.$myMessageTextBackground,
        padding: 10,
        marginBottom: 10,
        marginLeft: 10,
        marginRight: 10,
        borderRadius: 5
    },
    messages: {
        backgroundColor: globalStyles.$appBackgroundColor,
        display: 'flex',
        flexDirection: 'column'
    },
    userImage: {
        width:41,
        height:41,
        marginTop:2
    },
    text: {
        fontSize: 18,
        fontFamily: 'arial',
        color: globalStyles.$white
    }
});

export default class SingleMessage extends React.Component {
    render() {
        return (
            <View style={styles.messages}>
                { this.props.author === 'me' ? (
                    <View style={styles.myMessageRow}>
                        <AsyncImage source={require('./../../img/icons/user2.png')} style={styles.userImage} placeholderColor='#595959' />
                        <View style={styles.myMessageViewText}>
                            <Text style={styles.text}>{this.props.text}</Text>
                        </View>
                    </View>
                ) : (
                    <View style={styles.otherMessageRow}>
                        <AsyncImage source={require('./../../img/icons/user2.png')} style={styles.userImage} placeholderColor='#3a3a3a' />
                        <View style={styles.otherMessageViewText}>
                            <Text style={styles.text}>{this.props.text}</Text>
                        </View>
                    </View>
                ) }
            </View>
        );
    }
}

SingleMessage.propTypes = {
    text: PropTypes.string,
    author: PropTypes.string
}