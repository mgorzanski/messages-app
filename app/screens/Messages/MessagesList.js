import React from 'react';
import { StyleSheet, View, TouchableHighlight, Image } from 'react-native';
import { Container, Content, List } from 'native-base';
import Message from './Message';
import UserPanel from './UserPanel';
import * as globalStyles from './../../styles/globalStyles';
import PropTypes from 'prop-types';

export default class MessagesList extends React.Component {
    static navigationOptions = ({ navigation }) => ({
        title: "Messages",
        headerStyle: globalStyles.headerStyle,
        headerTitleStyle: globalStyles.headerTitleStyle,
        headerRight: (
            <View style={globalStyles.headerMultipleIcons}>
                <TouchableHighlight onPress={() =>
                    navigation.navigate('AddContact')
                } style={globalStyles.headerIcon}>
                    <Image 
                        source={require('./../../img/icons/search.png')}
                        style={globalStyles.icon}
                    />
                </TouchableHighlight>
                <TouchableHighlight onPress={() =>
                    navigation.navigate('AddContact')
                } style={globalStyles.headerIcon}>
                    <Image 
                        source={require('./../../img/icons/add.png')}
                        style={globalStyles.icon}
                    />
                </TouchableHighlight>
            </View>)
    });

    render() {
        return (
            <Container style={styles.home}>
                <Content>
                    <TouchableHighlight onPress={() => 
                        this.props.navigation.navigate('Profile')
                    }>
                        <UserPanel />
                    </TouchableHighlight>
                    <List>
                        <TouchableHighlight onPress={() =>
                            this.props.navigation.navigate('MessageThread')    
                        }>
                            <Message userName="Lorem Ipsum" message="Testowa wiadomość..." date="13:39" />
                        </TouchableHighlight>
                        <Message userName="Lorem Ipsum" message="Testowa wiadomość..." date="13:39" />
                        <Message userName="Lorem Ipsum" message="Testowa wiadomość..." date="13:39" />
                        <Message userName="Lorem Ipsum" message="Testowa wiadomość..." date="13:39" />
                        <Message userName="Lorem Ipsum" message="Testowa wiadomość..." date="13:39" />
                        <Message userName="Lorem Ipsum" message="Testowa wiadomość..." date="13:39" />
                    </List>
                </Content>
            </Container>
        );
    }
}

const styles = StyleSheet.create({
    home: {
        backgroundColor: globalStyles.$appBackgroundColor
    }
});

MessagesList.propTypes = {
    navigation: PropTypes.object.isRequired
};