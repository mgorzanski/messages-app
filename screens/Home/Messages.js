import React from 'react';
import { ScrollView } from 'react-native';
import Message from './Message';

export default class Messages extends React.Component {
    render() {
        return (
            <ScrollView>
                <Message userName="Lorem Ipsum" message="Testowa wiadomość..." date="11.03.2018 13:39" />
                <Message userName="Lorem Ipsum" message="Testowa wiadomość..." date="11.03.2018 13:39" />
                <Message userName="Lorem Ipsum" message="Testowa wiadomość..." date="11.03.2018 13:39" />
                <Message userName="Lorem Ipsum" message="Testowa wiadomość..." date="11.03.2018 13:39" />
                <Message userName="Lorem Ipsum" message="Testowa wiadomość..." date="11.03.2018 13:39" />
                <Message userName="Lorem Ipsum" message="Testowa wiadomość..." date="11.03.2018 13:39" />
            </ScrollView>
        );
    }
}