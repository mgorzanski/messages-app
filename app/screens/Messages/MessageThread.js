import React from 'react';
import { Container, Content } from 'native-base';
import { ScrollView } from 'react-native';
import { StyleSheet } from 'react-native';
import SingleMessage from './SingleMessage';
import { NavigationActions } from 'react-navigation';
import * as globalStyles from './../../styles/globalStyles';
import Icon from './../../utils/Icon';

const styles = StyleSheet.create({
    container: {
        backgroundColor: globalStyles.$appBackgroundColor,
        height: '100%'
    },
    thread: {
        flex: 1,
        margin: 10
    }
});

export default class MessageThread extends React.Component {
    static navigationOptions = {
            title: "Lorem Ipsum",
            headerStyle: globalStyles.headerStyle,
            headerTitleStyle: globalStyles.headerTitleStyle,
            headerRight: (<Icon family="MaterialIcons" name="more-vert" style={globalStyles.stackNavIcon} />),
            headerTintColor: globalStyles.$white
    };

    componentWillMount() {
        const setParamsAction = NavigationActions.setParams({
            params: { hideTabBar: true },
            key: 'tab-name'
        });
        this.props.navigation.dispatch(setParamsAction);
    }

    componentWillUnmount() {
        const setParamsAction = NavigationActions.setParams({
            params: { hideTabBar: false },
            key: 'tab-name'
        });
        this.props.navigation.dispatch(setParamsAction);
    }

    render() {
        return (
            <Container style={styles.container}>
                <ScrollView>
                    <Content style={styles.thread}>
                        <SingleMessage text="Testowa wiadomość" author="a" />
                        <SingleMessage text="..." author="me" />
                        <SingleMessage text="abcd" author="a" />
                        <SingleMessage text="Testowa wiadomość" author="me" />
                        <SingleMessage text="Testowa wiadomość" author="me" />
                    </Content>
                </ScrollView>
            </Container>
        );
    }
}