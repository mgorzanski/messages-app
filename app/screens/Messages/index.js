import React from 'react';
import { Image } from 'react-native';
import { StackNavigator } from 'react-navigation';
import MessagesList from './MessagesList';
import Profile from './../Profile';
import MessageThread from './MessageThread';
import * as globalStyles from './../../styles/globalStyles';
import getSlideFromRightTransition from 'react-navigation-slide-from-right-transition';

const Messages = StackNavigator({
    MessagesList: { screen: MessagesList },
    Profile: { screen: Profile },
    MessageThread: { screen: MessageThread }
}, {
    cardStyle: {
        opacity: 1
    },
    transitionConfig: getSlideFromRightTransition
});

const tabBarIcon = () => (<Image source={require('./../../img/icons/messages.png')} style={globalStyles.tabBarIcon} />);
Messages.navigationOptions = {
    tabBarLabel: '',
    tabBarIcon: tabBarIcon
}

export default Messages;