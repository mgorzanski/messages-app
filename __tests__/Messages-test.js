import React from 'react';
import Messages from './../screens/Home/Messages';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
    const tree = renderer.create(<Messages />).toJSON();
    expect(tree).toMatchSnapshot();
});