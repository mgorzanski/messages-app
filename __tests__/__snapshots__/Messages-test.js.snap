// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView>
  <View>
    <View
      style={
        Object {
          "backgroundColor": "#2d2d2d",
          "borderBottomColor": "#efefef",
          "borderBottomWidth": 0.25,
          "borderStyle": "solid",
          "display": "flex",
          "flexDirection": "row",
          "height": 75,
          "justifyContent": "space-between",
        }
      }
    >
      <Image
        source={
          Object {
            "process": [Function],
          }
        }
        style={
          Object {
            "height": 53,
            "marginLeft": 13,
            "marginTop": 10,
            "width": 53,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#2d2d2d",
          "borderBottomColor": "#efefef",
          "borderBottomWidth": 0.25,
          "borderStyle": "solid",
          "display": "flex",
          "flexDirection": "row",
          "height": 75,
          "justifyContent": "space-between",
        }
      }
    >
      <Image
        source={
          Object {
            "process": [Function],
          }
        }
        style={
          Object {
            "height": 53,
            "marginLeft": 13,
            "marginTop": 10,
            "width": 53,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#2d2d2d",
          "borderBottomColor": "#efefef",
          "borderBottomWidth": 0.25,
          "borderStyle": "solid",
          "display": "flex",
          "flexDirection": "row",
          "height": 75,
          "justifyContent": "space-between",
        }
      }
    >
      <Image
        source={
          Object {
            "process": [Function],
          }
        }
        style={
          Object {
            "height": 53,
            "marginLeft": 13,
            "marginTop": 10,
            "width": 53,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#2d2d2d",
          "borderBottomColor": "#efefef",
          "borderBottomWidth": 0.25,
          "borderStyle": "solid",
          "display": "flex",
          "flexDirection": "row",
          "height": 75,
          "justifyContent": "space-between",
        }
      }
    >
      <Image
        source={
          Object {
            "process": [Function],
          }
        }
        style={
          Object {
            "height": 53,
            "marginLeft": 13,
            "marginTop": 10,
            "width": 53,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#2d2d2d",
          "borderBottomColor": "#efefef",
          "borderBottomWidth": 0.25,
          "borderStyle": "solid",
          "display": "flex",
          "flexDirection": "row",
          "height": 75,
          "justifyContent": "space-between",
        }
      }
    >
      <Image
        source={
          Object {
            "process": [Function],
          }
        }
        style={
          Object {
            "height": 53,
            "marginLeft": 13,
            "marginTop": 10,
            "width": 53,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#2d2d2d",
          "borderBottomColor": "#efefef",
          "borderBottomWidth": 0.25,
          "borderStyle": "solid",
          "display": "flex",
          "flexDirection": "row",
          "height": 75,
          "justifyContent": "space-between",
        }
      }
    >
      <Image
        source={
          Object {
            "process": [Function],
          }
        }
        style={
          Object {
            "height": 53,
            "marginLeft": 13,
            "marginTop": 10,
            "width": 53,
          }
        }
      />
    </View>
  </View>
</RCTScrollView>
`;
